impl/src/main/java/org/jboss/weld/executor/AbstractExecutorServices.java

```diff
@@ -55,7 +55,8 @@ public ScheduledExecutorService getTimerExecutor() {
     @Override
     public <T> List<Future<T>> invokeAllAndCheckForExceptions(Collection<? extends Callable<T>> tasks) {
         try {
-            return checkForExceptions(getTaskExecutor().invokeAll(tasks));
+            // the cast is needed to compile this expression in JDK 8, works without it on JDK 10+
+            return checkForExceptions(getTaskExecutor().invokeAll((Collection<? extends Callable<T>>)wrap(tasks)));
```

Comment:
This cast should be harmless as it is the exact same type as is returned by `wrap()` function.
However, I found no one-liner solution to avoid this and still be able to compile on JDK 8.

Reply:
Np, we already have similar casts in the codebase here and there...